### PR TITLE
refs #2092; port copySessionDescription to all relevant examples; tin…

### PR DESCRIPTION
**Pull request by r-novel**
…y fixes codestyle in api.go file;

#### Description
I've updated all relevant jsfiddles in the example directory.

Tiny fixes codestyle in api.go. Move default initialize before executing option callbacks.

refs #2092 ticket;

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -70,8 +63,12 @@ func WithSettingEngine(s SettingEngine) func(a *API) {
 
 // WithInterceptorRegistry allows providing Interceptors to the API.
 // Settings should not be changed after passing the registry to an API.
-func WithInterceptorRegistry(interceptorRegistry *interceptor.Registry) func(a *API) {
+func WithInterceptorRegistry(ir *interceptor.Registry) func(a *API) {
```
Comment (Sean-Der) [inline api.go:66]: Can you add a test to confirm the new behavior? `WithSettingRegistry` that is nil shouldn't segfault

File: api.go [hunk 2]
```diff
@@ -23,28 +23,21 @@ type API struct {
...
+		settingEngine:       &SettingEngine{},
+		mediaEngine:         &MediaEngine{},
+		interceptorRegistry: &interceptor.Registry{},
+	}
 
 	for _, o := range options {
 		o(a)
 	}
 
-	if a.settingEngine == nil {
```
Comment (Sean-Der) [inline api.go:32]: Can you also add a `ni` check to `WithSettingEngine` to match WithMediaEngine + WithInterceptorRegistry.

**Comment by Sean-Der** [comment]
Great work @r-novel, awesome job!

Mind sending over the `With..` as a separate change. Will approve and merge right away :) glad to make the code more consistent and cleaner.

Would you mind also updating the README.md of the examples you updated?

File: api.go (hunk 2 above)
Comment (r-novel) [reply api.go:32]: Done! But I think this nil check is not necessary because in WithSettingEngine function we pass SettingEngine parameter by value - not by pointer.

**Comment by r-novel** [comment]
Hello, @Sean-Der ! Thank you so much for your code review. I've added unit tests and updated README files.

**Comment by Sean-Der** [comment]
Great job @r-novel merging!

Some of the examples didn't have `log` defined so I switched to `console.log`. I also updated all the jsfiddle links with your new code, thanks for doing this :)

**Review by Sean-Der — APPROVED** [review]
